DependencyInjection/Compiler/AddProvidersPass.php

```diff
@@ -28,6 +31,10 @@ public function process(ContainerBuilder $container)
             // when using only one (the default case as the bundle defines one provider)
             $container->setAlias('knp_menu.menu_provider', (string) reset($providers));
         } else {
+            if (class_exists(IteratorArgument::class)) {
```

Comment:
is this symfony 3.3+ only? want to add a comment so we know we can drop the exists check when we drop support for older versions of symfony?

Reply:
yep, the class was added in 3.3

Reply:
And this is part of the WIP commit. I'm not done yet on these.